program/rust/src/processor/add_publisher.rs

```diff
@@ -73,14 +83,149 @@ pub fn add_publisher(
         }
     }
 
-    let current_index: usize = try_convert(price_data.num_)?;
+    let mut current_index: usize = try_convert(price_data.num_)?;
     sol_memset(
         bytes_of_mut(&mut price_data.comp_[current_index]),
         0,
         size_of::<PriceComponent>(),
     );
     price_data.comp_[current_index].pub_ = cmd_args.publisher;
+
+    // Shift the element back to keep the publishers components sorted.
+    while current_index > 0
+        && price_data.comp_[current_index].pub_ < price_data.comp_[current_index - 1].pub_
+    {
+        price_data.comp_.swap(current_index, current_index - 1);
+        current_index -= 1;
+    }
```

Comment:
minor comment: imo, we should call `sort_price_comps` here as well, because the insertion code above isn't tested by the quickcheck stuff below. We don't add publishers a lot so we don't care about optimizing CU use.

Reply:
(feel free to ignore this. I'm only saying it because it feels like the maximally defensive implementation)

Reply:
did that. removed the `inline` hint to make sure code size doesn't get larger (by sacrificing a bit of CU)